### DISCUSSION TITLE
View only mode

### DISCUSSION THREAD

**Discussion by swingthrough** [jgraph/drawio#4178] [Q&A] [answered]
We have a drawio diagram MS Teams app, and when I opened it, I noticed I can make changes. I started searching for a setting that switches me into view mode so I don't accidentally edit the contents, however I was unable to find such a setting.

I found this discussion, but it relates to desktop app:
https://github.com/jgraph/drawio-desktop/discussions/1380

There's a comment there:

> That doesn't really make sense for the desktop version. You'd use the online version for sharing.

Well first of all I don't understand how it doesn't make sense for the desktop version either, but anyway, I went to the online version and searched for a setting like this to no avail.

What I would expect is some kind of quickly accessible switch called  "view mode" or something in that sense, that would lock the diagram and I wouldn't have to think about being so careful to not accidentally edit the contents.

I actually had a similar issue with Figma, which also doesn't have such a feature, but there is at least this thread:
https://forum.figma.com/t/switch-to-read-only-mode/29558

So people are frustrated about the fact that such a feature does not exist - it might add some context and relevance to my question.

I don't want to subtract my privileges for editing (which is how I "solved" this issue in Figma on projects at work where I didn't need edit privileges), I just want to be able to quickly switch back and forth. And I want to be able to do it in my own diagrams as well.

A feature like this explicitly separates the workflow of "I now only need the diagram for reference to implement something based on it". There is no need to have edit switched on, in fact it is dangerous as I might mess something up when not careful.

So basically the questions are:
1. Is there or is it planned to have such a feature for **desktop** app?
2. Is there or is it planned to have such a feature for **online** app?
3. Is there or is it planned to have such a feature for **MS Teams** app?

**Comment by davidjgraph** [comment, marked as answer]
1. No plans.
2. If you store the diagram somewhere that supports read only permissions (Google Drive) then it uses the permissions from the authenication system.
3. No plans. I can see the idea, but the problem is more the amount of ideas there are as feature request and the number of people available to implement them.
